### PR TITLE
Add theme reset section

### LINKED ISSUE #494: Ability to reset theme style settings
## What problem does this address?

A common workflow for updating theme settings using Create Block Theme is

1. Load an existing base theme (Twenty Twenty Four is currently a common choice)
2. Edit Global Styles
3. Persist those to theme.json, using Create Block theme
4. Export the theme files

One thing you can't do in this workflow (without manually editing theme.json) is to reset or remove a theme's styles, without overriding them.

We should have a way to do this in the UI, at least through [a theme.json editor](https://github.com/WordPress/create-block-theme/issues/491), but ideally in the Global Styles UI.

### LINKED ISSUE #686: Ability to reset all settings in one click
This idea is from @carolinan from this GB issue: https://github.com/WordPress/gutenberg/issues/54958.

> I often find that I need to go to three places to reset changes:
>
> 1. The template management screen
> 2. The template part management screen
> 3. The styles sidebar
> 
> On the template and part management screens, I miss an option to reset all my items with one click.
> 
> In general I miss an option to reset all my user made changes, so both custom CSS, block style changes, template part markup changes, and template markup changes.
> 
> The option for me is to install and use a third party reset plugin, but those delete all my content and deactivate my current theme and plugins.

Related to https://github.com/WordPress/create-block-theme/issues/494.

### PR DESCRIPTION
**Pull request by matiasbenedetto**
## What?
Add theme reset section

## Why?
Closes: https://github.com/WordPress/create-block-theme/issues/686 https://github.com/WordPress/create-block-theme/issues/494

## Screencast
[Screencast from 09-07-24 16:37:50.webm](https://github.com/WordPress/create-block-theme/assets/1310626/1a8c48ad-2c99-44df-a8bc-5fa791766522)

## Testing instructions:
1. make changes to theme templates using the site editor.
2. make changes to theme template parts using the site editor.
3. make changes to theme global styles using the site editor.
4. save the changes in the editor.
5. use the CBT reset functionality and check that the theme customization was erased.

### CODE REVIEW

File: src/editor-sidebar/reset-theme.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+	};
+
+	const handleResetTheme = async () => {
+		try {
+			await resetTheme( preferences );
+			toggleConfirmDialog();
+			// eslint-disable-next-line no-alert
+			window.alert(
+				__(
+					'Theme reseted successfully. The editor will now reload.',
```
Comment (mikachan) [inline src/editor-sidebar/reset-theme.js]:
```suggestion
					'Theme reset successfully. The editor will now reload.',
```

File: src/editor-sidebar/reset-theme.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+					'create-block-theme'
+				)
+			);
+			window.location.reload();
+		} catch ( error ) {
+			createErrorNotice(
+				__(
+					'An error occurred while resetting theme.',
```
Comment (mikachan) [inline src/editor-sidebar/reset-theme.js]:
```suggestion
					'An error occurred while resetting the theme.',
```

File: src/editor-sidebar/reset-theme.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+							'create-block-theme'
+						) }
+						checked={ preferences.resetTemplateParts }
+						onChange={ () =>
+							handleTogglePreference( 'resetTemplateParts' )
+						}
+					/>
+
+					<Button
+						text={ __( 'Reset theme', 'create-block-theme' ) }
```
Comment (mikachan) [inline src/editor-sidebar/reset-theme.js]:
```suggestion
						text={ __( 'Reset Theme', 'create-block-theme' ) }
```

File: includes/class-create-block-theme-api.php [hunk 4]
```diff
@@ -144,6 +144,18 @@ public function register_rest_routes() {
...
 		);
+		register_rest_route(
+			'create-block-theme/v1',
+			'/reset-theme',
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => array( $this, 'rest_reset_theme' ),
+				// 'permission_callback' => function () {
+				// 	return current_user_can( 'edit_theme_options' );
+				// },
```
Comment (mikachan) [inline includes/class-create-block-theme-api.php]: Can we remove this comment or should we add the permission check?

**Review by mikachan — COMMENTED** [review]
This is testing well for me, nice work 🙌  I've left some minor comments and tiny wording suggestions.

File: includes/class-create-block-theme-api.php (hunk 4 above)
Comment (matiasbenedetto) [reply includes/class-create-block-theme-api.php]: thanks for noticing that, the comment was for testing something locally. It wasn't plan to be commited. I removed it now:  [`8ca3269` (#687)](https://github.com/WordPress/create-block-theme/pull/687/commits/8ca3269c2a42d47c973889017a20632164d87b43)

**Review by mikachan — APPROVED** [review]
Thanks! I think this is ready to bring in 🎉

**Pull request merged**